Restore downstream tests with mambabuild

### Comment:

Both #110 and #109 are both failing on the downstreams due to conda errors like:

```
TEST START: /home/conda/feedstock_root/build_artifacts/pkg_cache/jupyterlab-3.5.2-pyhd8ed1ab_0.conda
Copying package to conda-build croot.  No packages otherwise alongside yours will be available unless you specify -c local.  To avoid this warning, your package must reside in a channel structure with platform-subfolders.  See more info on what a valid channel is at https://conda.io/docs/user-guide/tasks/create-custom-channels.html
WARNING:conda_build.build:Copying package to conda-build croot.  No packages otherwise alongside yours will be available unless you specify -c local.  To avoid this warning, your package must reside in a channel structure with platform-subfolders.  See more info on what a valid channel is at https://conda.io/docs/user-guide/tasks/create-custom-channels.html
Reloading output folder: /home/conda/feedstock_root/build_artifacts
Traceback (most recent call last):
  File "/opt/conda/bin/conda-mambabuild", line 10, in <module>
    sys.exit(main())
  File "/opt/conda/lib/python3.10/site-packages/boa/cli/mambabuild.py", line 256, in main
    call_conda_build(action, config)
  File "/opt/conda/lib/python3.10/site-packages/boa/cli/mambabuild.py", line 228, in call_conda_build
    result = api.build(
  File "/opt/conda/lib/python3.10/site-packages/conda_build/api.py", line 180, in build
    return build_tree(
  File "/opt/conda/lib/python3.10/site-packages/conda_build/build.py", line 3149, in build_tree
    test(list(local_file.values())[0][0],
  File "/opt/conda/lib/python3.10/site-packages/conda_build/build.py", line 2874, in test
    actions = environ.get_install_actions(metadata.config.test_prefix,
  File "/opt/conda/lib/python3.10/site-packages/boa/cli/mambabuild.py", line 124, in mamba_get_install_actions
    solver = _get_solver(channel_urls, subdir, output_folder)
  File "/opt/conda/lib/python3.10/site-packages/boa/cli/mambabuild.py", line 100, in _get_solver
    solver.replace_channels()
  File "/opt/conda/lib/python3.10/site-packages/boa/core/solver.py", line 177, in replace_channels
    self.local_repos[channelstr] = libmambapy.Repo(
RuntimeError: Could not read JSON repodata file (/home/conda/feedstock_root/build_artifacts/pkg_cache/cache/cf8a841c.json) parse error line 34: subdir mismatch
```

Probably going to just ship without testing the downstreams, but would be good to get this back.

```
    jupyterlab_server:             2.16.5-pyhd8ed1ab_0       local      
    openapi-core:                  0.14.2-pyhd8ed1ab_0       conda-forge
    openapi-schema-validator:      0.3.4-pyhd8ed1ab_0        conda-forge
    openapi-spec-validator:        0.5.1-pyhd8ed1ab_0        conda-forge
```

Then

```
E   ImportError: cannot import name 'default_handlers' from 'openapi_spec_validator' ($PREFIX/lib/python3.11/site-packages/openapi_spec_validator/__init__.py)
```

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -10,7 +10,7 @@ source:
 
 build:
   noarch: python
-  number: 0
+  number: 1
   script:
     - {{ PYTHON }} -m pip install . --no-deps --ignore-installed --no-cache-dir -vvv
   entry_points:
@@ -68,8 +68,7 @@ test:
     - jupyter_server
   downstreams:
     - jupyterlab >=3.2
-    # TODO: fix "ImportError: cannot import name 'default_handlers' from 'openapi_spec_validator'"
-    # - jupyterlab_server >=2.14.0
+    - jupyterlab_server >=2.17.0
 
 about:
   home: https://jupyter-server.readthedocs.io

